src/config/system_stats/network_down.rs

```diff
@@ -0,0 +1,22 @@
+use lazy_static::lazy_static;
+use std::sync::{Arc, Mutex};
+use sysinfo::{NetworkExt, NetworksExt, RefreshKind, System, SystemExt};
+
+lazy_static! {
+    static ref SYSTEM: Arc<Mutex<System>> = Arc::new(Mutex::new(System::new_with_specifics(
+        RefreshKind::new().with_networks_list()
+    )));
+}
+
+pub fn get_down() -> f32 {
+    let sys = SYSTEM.clone();
+    let mut c = sys.lock().unwrap();
+    c.refresh_networks();
+    let interfaces: u64 = c
+        .get_networks()
+        .iter()
+        .filter(|a| !a.0.starts_with("docker") || !a.0.starts_with("lo"))
```

Comment:
bruh please tell me there is a way to do this more cleanly,... like some way to detect internal (logical?) interfaces generically,... this is really ewwwwwwwww

Reply:
well that's the entire point, is it not? 
For real tho, `route` can display which interface is being used to communicate with the internet. I can look into this. No promises tho

Reply:
It seems like /proc/self/net/route has all of the routes, and by that extension also only all interfaces that are connected to the internet, or a network large enough, where this would be necessary.

So i could parse this and then only use the interfaces found in that file.

Reply:
if that's the best way, then sure. I'd have hoped that the library provides some direct functionality for that, but if not then so be it, parsing that file sounds like a good alternative. Make sure that you keep macos compat in mind though (i'd be _fine_ with having _less_ functionality on macos - i.e. by just not checking for only active network interfaces there or whatever (although doing the same checks for macos would be optimal, of course), but it's important that it at least doesn't crash :D)